src/main/java/com/softserve/academy/spaced/repetition/service/impl/AccountServiceImpl.java

```diff
@@ -36,43 +30,24 @@
     @Autowired
     private UserService userService;
     @Autowired
-    private NumberOfPostponedDaysValidator numberOfPostponedDaysValidator;
-    @Autowired
     private MailService mailService;
     @Autowired
     private PasswordEncoder passwordEncoder;
-    @Autowired
-    private MessageSource messageSource;
-    private final Locale locale = LocaleContextHolder.getLocale();
-
-    @Override
-    public void updateAccount(Account account) {
-        accountRepository.save(account);
-    }
 
     @Override
     @Transactional
-    public LearningRegime getLearningRegime() throws NotAuthorisedUserException {
-        return userService.getAuthorizedUser().getAccount().getLearningRegime();
+    public Account updateAccountDetails(Account acc) throws NotAuthorisedUserException {
+        Account account = userService.getAuthorizedUser().getAccount();
+        account.setRememberingLevels(acc.getRememberingLevels());
+        account.setCardsNumber(acc.getCardsNumber());
+        account.setLearningRegime(acc.getLearningRegime());
+        return accountRepository.save(account);
```

Comment:
If i'm not mistaken, we not need to save object, that we get from DB by some repository.